src/debugpy/server/adapters.py

```diff
@@ -346,6 +350,40 @@ def next_request(self, request: Request):
         self._tracer.step_over(thread)
         return {}
 
+    def gotoTargets_request(self, request: Request) -> dict:
+        path = request("source", dict)["path"]
```

Comment:
It should also check that "path" is a string. This will do all the necessary validation:
```py
source = request("source", json.object())  # if missing you get {}
path = source("path", str)
```

Also, this is not the only place that will have to deal with "source", since it's a recurring thing in DAP, both incoming and outgoing (e.g. take a look at `setBreakpoints_request`). For outgoing, there's class `Source` in `debug.server.tracing`; perhaps it should get a `from_json` class method?

Long-term I think it's best to drop this approach to validating JSON at point of use entirely, and instead parse and validate messages before they get dispatched to handlers. The repo with the DAP spec also has a complete JSON schema for it, from which strongly typed classes could be generated (this is what pydevd does).